### PR TITLE
Change document URL to wiki when searching on it

### LINKED ISSUE #5809: New frontend - Search from wiki.d.m.o: maintain consistency for links/URL domain
When using the search from wiki.d.m.o, the list of links points to d.m.o.

This is not helpful when trying to open the several pages for edition in tabs (ex. searching for a macro to delete and wanting to open batches of pages containing the macro).

On the other side, when using the search from d.m.o, the links correctly point to d.m.o

### PR DESCRIPTION
**Pull request by Gregoor**
Fixes #5809

Changes document URLs in search to a wiki document URl when searching through wiki

### CODE REVIEW

File: kuma/search/jinja2/search/results.html [hunk 1]
```diff
@@ -22,15 +22,16 @@
...
     <div class="column-container">
       <div class="column-1">
           {% include 'includes/icons/file-icons/plain-text.svg' %}
       </div>
       <div class="column-5 result-list-item">
-        <h4><a href="{{ doc.url }}"{% if index == 1 %} tabindex="1"{% endif %}>{{ doc.title }}</a></h4>
+        <h4><a href="{{ doc_url }}"{% if index == 1 %} tabindex="1"{% endif %}>{{ doc.title }}</a></h4>
         <p>{{ doc.excerpt|safe }}</p>
-        <p class="search-meta"><a href="{{ doc.url }}">{{ doc.url|replace('https://', '') }}</a>
+        <p class="search-meta"><a href="{{ doc_url }}">{{ doc_url|replace('https://', '') }}</a>
```
Comment (peterbe) [inline kuma/search/jinja2/search/results.html]:
```suggestion
        <p class="search-meta"><a href="{{ doc_url }}">{{ doc_url }}</a>
```
It's never an external URL any more.

File: kuma/search/jinja2/search/results.html [hunk 2]
```diff
@@ -22,15 +22,16 @@
 {% block bodyclass %}search-results search-results-no-demo search-navigator-flush{% endblock %}
 
 {% macro doc_block(doc, index) -%}
+  {% set doc_url = wiki_url(doc.slug) if wiki else doc.url %}
```
Comment (peterbe) [inline kuma/search/jinja2/search/results.html]:
```suggestion
  {% set doc_url = wiki_url(doc.slug) %}
```
I don't think it's ever NOT `wiki`. The React site (aka. the not-wiki) uses `search-results-page.jsx` (or whatever it's called). 
I still like the idea making a variable once.

**Comment by Gregoor** [comment]
> 
> 
> This change only solves it for the wiki. E.g....
> Screen Shot 2019-09-30 at 12 30 46 PM
> 
> So either make the necessary changes to `search-results-page.jsx` or change the PR description to `Part of #5809`

I'm a bit confused about that one. How does one render React search results for the wiki? Do we have something like `beta.wiki.d.m.o`?

**Comment by peterbe** [comment]
UGh! Ignore me. I confused this for another *issue* so when I reviewed I must have had that in mind.

**Comment by Gregoor** [comment]
Happens! I made changes according to the other two comments, so I hope that covers it all :)

**Pull request merged**